pkg/common/utils/utils.go

```diff
@@ -82,24 +82,30 @@ func GetQueueNameFromPod(pod *v1.Pod) string {
 }
 
 func GetApplicationIDFromPod(pod *v1.Pod) (string, error) {
-	for name, value := range pod.Labels {
-		// if a pod for spark already provided appID, reuse it
-		if name == constants.SparkLabelAppID {
+	// application ID can be defined in annotations
+	for name, value := range pod.Annotations {
+		if name == constants.LabelApplicationID {
+			return value, nil
+		}
```

Comment:
these lines should not be here, this is already covered by line 98 - 100, I think we can remove them

Reply:
Is not the same. Here we check the annotations and in lines 89-100 we check the labels

Reply:
I don't think we need to check `constants.LabelApplicationID` here in this case.
When we ask users to specify annotations, that should follow the form: "yunikorn.apache.org/app-id".

Reply:
I removed constants.LabelApplicationID from the annotations part and also renamed the SparkAnnotationAppID to AnnotationApplicationID, since we will use it for other kind of apps as well.

Reply:
Thx